app/helpers/assessment_autograde_core.rb

```diff
@@ -412,10 +412,26 @@ def saveAutograde(submissions, feedback)
       @autograde_prop = @assessment.autograder
 
       # Record each of the scores extracted from the autoresult
-      scores.keys.each do |key|
-        problem = @assessment.problems.find_by(name: key)
-        raise AutogradeError.new("Problem \"" + key + "\" not found.") unless problem
-        submissions.each do |submission|
+      submissions.each do |submission|
+
+        # If modifySubmissionScore is overridden, use that to modify the score
+        # Provide it with the scores and previous submissions
+        if @assessment.overwrites_method?(:modifySubmissionScores)
+          
+          # Get previous submissions of the same assessment by the user, excluding the current submission
+          previous_submissions = Submission.where(course_user_datum_id: submission.course_user_datum_id, 
+                                 assessment_id: @assessment.id).where.not(id: submission.id)
+
+          begin
+            scores = @assessment.config_module.modifySubmissionScores(scores, previous_submissions)
+          rescue StandardError => e
+            raise AutogradeError.new("Error in modifySubmissionScores: " + e.message)
+          end
+        end
+
+        scores.keys.each do |key|
```

Comment:
note that I flipped the loop (it used to be scores being in the outer loop, and the submissions in the inner loop).

Because I want the scores to be modified per submission